src/data/moves/move-attrs/ability-give-attr.ts

```diff
@@ -33,4 +35,22 @@ export class AbilityGiveAttr extends MoveEffectAttr {
       && !target.getAbility().hasAttrFlag(AbAttrFlag.UNSUPPRESSABLE_ABILITY)
       && user.getAbility().id !== target.getAbility().id;
   }
+
+  /**
+   * If the user has a {@link DETRIMENTAL_ABILITIES | detrimental ability}, or
+   * the target has a {@link HIGH_VALUE_ABILITIES | high-value ability},
+   * grants (+1) with a 50% chance of additional (+1) to effect score.
+   */
+  override getEffectScore(user: EnemyPokemon, target: Pokemon, _move: Move): number {
+    if (HIGH_VALUE_ABILITIES.includes(user.getAbility().id)) {
+      return -5;
```

Comment:
This `-5` number should probably be a descriptive constant value

and not a _magic number_